### PR TITLE
refactor: Replace white & black colors with theme aware ones

### PR DESCRIPTION
**Pull request by luchillo17**
## Hey, I just made a Pull Request!

Based on discussions on #15677 it was decided we'll change black & white colors to be theme aware, and if possible the colors that are close to the `theme.palette.status.*` colors.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by luchillo17** [comment]
I saw some files with `*Chip` in the name and wanted to change the colors to use `theme.palette.status.*` ones, but couldn't properly type the theme variable to use `BackstageTheme`, still looking how to do this but from the typings code I think `withStyles` was not made to allow custom themes, or at least not in the way we use the custom theme, maybe if we augmented the original theme instead of extending it as `BackstageTheme`...

**Comment by luchillo17** [comment]
I'm a bit confused with the failing test, it seems the theme in which the test are running is not a `BackstageTheme` but a normal MUI v4 theme...

File: plugins/azure-sites/src/components/AzureSitesOverviewTableComponent/AzureSitesOverviewTable.test.tsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ describe('AzureSitesOverviewWidget', () => {
   });
 
   it('should display an overview table with the data from the requests', async () => {
-    const rendered = render(
+    const rendered = await renderInTestApp(
```
Comment (jamieklassen) [inline plugins/azure-sites/src/components/AzureSitesOverviewTableComponent/AzureSitesOverviewTable.test.tsx:69]: looks like switching away from using `render` here [makes the import unused](https://github.com/backstage/backstage/actions/runs/4245218628/jobs/7380437688#step:7:10)

Comment (luchillo17) [reply plugins/azure-sites/src/components/AzureSitesOverviewTableComponent/AzureSitesOverviewTable.test.tsx:69]: Indeed, will fix.

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Do you think this is a good middle ground where we can break for now and see if any issues pop up?

I do think overall that it would be worth while investigating gaps in our current theme palette and try to make as much as possible themeable, but it does require a quite careful approach and takes more time.

**Pull request merged**